test/core/primitives/sector_file/sector_file_test.cpp

```diff
@@ -240,9 +241,57 @@ namespace fc::primitives::sector_file {
     EXPECT_OUTCOME_TRUE(
         sector_size,
         fc::primitives::sector::getSectorSize(min_seal_proof_type));
-    uint64_t result =
+    const uint64_t result =
         kOverheadSeal.at(file_type) * sector_size / kOverheadDenominator;
     EXPECT_OUTCOME_TRUE(seal_size, sealSpaceUse(file_type, sector_size));
     ASSERT_EQ(result, seal_size);
   }
+
+  TEST_F(SectorFileTest, checkFlag) {
+    const PieceData tempPieceData = PieceData();
+    const bool isNull = tempPieceData.isNullData();
+    EXPECT_EQ(isNull, true);
+  }
+
+  /*
+   * TODO (@Markuu-s) add tests: CannotMoveCursor and checkAssertGetFd
+   * Problem: ASAN-build doesn't pass these tests
+   */
+
+  TEST_F(SectorFileTest, WriteNullPiece) {
+    EXPECT_OUTCOME_TRUE(
+        sector_size,
+        fc::primitives::sector::getSectorSize(min_seal_proof_type));
+    const SectorId sector{
+        .miner = 1,
+        .sector = 1,
+    };
+
+    const PaddedPieceSize piece_size(128);
+
+    EXPECT_OUTCOME_TRUE(result_cid,
+                        fc::sector_storage::zerocomm::getZeroPieceCommitment(
+                            piece_size.unpadded()));
+    EXPECT_OUTCOME_TRUE(
+        file,
+        SectorFile::createFile(
+            (base_path / fc::primitives::sector_file::sectorName(sector))
+                .string(),
+            PaddedPieceSize(sector_size)));
+
+    EXPECT_OUTCOME_TRUE(
+        piece_info,
+        file->write(PieceData(), 0, piece_size, min_seal_proof_type));
+
+    ASSERT_EQ(piece_info->size, piece_size);
+    EXPECT_EQ(piece_info->cid, result_cid);
+    EXPECT_OUTCOME_EQ(file->hasAllocated(0, piece_size.unpadded()), true);
+
+    EXPECT_OUTCOME_TRUE(cid,
+                        fc::sector_storage::zerocomm::getZeroPieceCommitment(
+                            piece_size.unpadded()));
+
+    EXPECT_EQ(piece_info->cid, cid);
```

Comment:
Call `file->read()` and check that your piece is zero-fill

Reply:
Fixed